tests/t0123_gateway_json_cbor_test.go

```diff
@@ -1,13 +1,859 @@
+		  test_should_contain "501 Not Implemented" curl_output &&
+		  test_should_contain "reading IPLD Kinds other than Links (CBOR Tag 42) is not implemented" curl_output
+		  '
+		*/
+		{
+			Name: "GET DAG-JSON traversal returns 501 if there is path remainder",
+			Request: Request().
+				Path("ipfs/%s/foo", dagJSONTraversalCID).
+				Query("format", "dag-json"),
+			Response: Expect().
+				Status(501).
+				Body(Contains("reading IPLD Kinds other than Links (CBOR Tag 42) is not implemented")),
+		},
+		/**
+		  test_expect_success "GET DAG-JSON traverses multiple links" '
+		  curl -s "http://127.0.0.1:$GWAY_PORT/ipfs/$DAG_JSON_TRAVERSAL_CID/foo/link/bar?format=dag-json" > curl_output 2>&1 &&
+		  jq --sort-keys . curl_output > actual &&
+		  echo "{ \"hello\": \"this is not a link\" }" | jq --sort-keys . > expected &&
+		  test_cmp expected actual
+		  '
+		*/
+		{
+			Name: "GET DAG-JSON traverses multiple links",
+			Request: Request().
+				Path("ipfs/%s/foo/link/bar", dagJSONTraversalCID).
+				Query("format", "dag-json"),
+			Response: Expect().
+				Status(200).
+				Body(
+					// TODO: I like that this text is readable and easy to understand.
+					// 		 but we might prefer matching abstract values, something like "IsJSONEqual(someFixture.formatedAsJSON))"
+					IsJSONEqual([]byte(`{"hello": "this is not a link"}`)),
+				),
+		},
+		/**
+		  test_expect_success "GET DAG-CBOR traversal returns 501 if there is path remainder" '
+		  curl -sD - "http://127.0.0.1:$GWAY_PORT/ipfs/$DAG_CBOR_TRAVERSAL_CID/foo?format=dag-cbor" > curl_output 2>&1 &&
+		  test_should_contain "501 Not Implemented" curl_output &&
+		  test_should_contain "reading IPLD Kinds other than Links (CBOR Tag 42) is not implemented" curl_output
+		  '
+		*/
+		{
+			Name: "GET DAG-CBOR traversal returns 501 if there is path remainder",
+			Request: Request().
+				Path("ipfs/%s/foo", dagCBORTraversalCID).
+				Query("format", "dag-cbor"),
+			Response: Expect().
+				Status(501).
+				Body(Contains("reading IPLD Kinds other than Links (CBOR Tag 42) is not implemented")),
+		},
+		/**
+		  test_expect_success "GET DAG-CBOR traverses multiple links" '
+		  curl -s "http://127.0.0.1:$GWAY_PORT/ipfs/$DAG_CBOR_TRAVERSAL_CID/foo/link/bar?format=dag-json" > curl_output 2>&1 &&
+		  jq --sort-keys . curl_output > actual &&
+		  echo "{ \"hello\": \"this is not a link\" }" | jq --sort-keys . > expected &&
+		  test_cmp expected actual
+		  '
+		*/
+		{
+			Name: "GET DAG-CBOR traverses multiple links",
+			Request: Request().
+				Path("ipfs/%s/foo/link/bar", dagCBORTraversalCID).
+				Query("format", "dag-json"),
+			Response: Expect().
+				Status(200).
+				Body(
+					// TODO: I like that this text is readable and easy to understand.
+					// 		 but we might prefer matching abstract values, something like "IsJSONEqual(someFixture.formatedAsJSON))"
+					IsJSONEqual([]byte(`{"hello": "this is not a link"}`)),
```

Comment:
@galargh I'll keep as is, sounds good?

Reply:
Would be nice if we could retrieve it from the car, but sounds good to me with the TODO 👍